.github/workflows/build.yml

```diff
@@ -248,14 +260,40 @@ jobs:
             exit 0
           fi
 
-          echo "Running integration tests with tags: $TEST_SUITE_TAGS (empty means everything)"
+          echo "Running integration tests with tags: $TEST_SUITE_TAGS (empty means every test not specifically tagged)"
 
           export TEST_SUITE_TAGS="$TEST_SUITE_TAGS"
+          TIMEOUT=10m
+          if [[ "$TEST_SUITE_TAGS" == "all" ]]; then
+            TIMEOUT=90m
+          fi
+          set +e
+          SHELL='' go test -timeout $TIMEOUT -v `go list ./... | grep integration` &> build/integration-tests.out
+          CODE=$?
+          cat build/integration-tests.out | gotestfmt
+          exit $CODE
+        env:
+          INTEGRATION_TEST_USERNAME: ${{ secrets.INTEGRATION_TEST_USERNAME }}
+          INTEGRATION_TEST_PASSWORD: ${{ secrets.INTEGRATION_TEST_PASSWORD }}
+          INTEGRATION_TEST_TOKEN: ${{ secrets.INTEGRATION_TEST_TOKEN }}
+          PLATFORM_API_TOKEN: ${{ secrets.PLATFORM_API_TOKEN }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+      - # === Nightly Integration Tests ===
+        name: Nightly Integration Tests
+        id: nightly_integration_tests
+        if: github.event_name == 'schedule'
```

Comment:
Don't need a separate job, just use this conditional inside the script logic for the job above.

Reply:
Ok.